File: fix_modals.py
import os

def fix_modals(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(".dart"):
                filepath = os.path.join(root, file)
                with open(filepath, "r", encoding="utf-8") as f:
                    content = f.read()
                
                if "showModalBottomSheet(" in content and "useRootNavigator: true" not in content:
                    new_content = content.replace("showModalBottomSheet(\n      context: context,", "showModalBottomSheet(\n      context: context,\n      useRootNavigator: true,")
                    new_content = new_content.replace("showModalBottomSheet(\n    context: context,", "showModalBottomSheet(\n    context: context,\n    useRootNavigator: true,")
                    
                    if new_content != content:
                        with open(filepath, "w", encoding="utf-8") as f:
                            f.write(new_content)
                        print(f"Fixed {filepath}")

fix_modals("lib")

File: fix_modals.dart
import 'package:flutter_riverpod/flutter_riverpod.dart'; // Just checking valid syntax? Wait, this is a scratch script

import 'dart:io';

void main() {
  final libDir = Directory('lib');
  final files = libDir.listSync(recursive: true).whereType<File>().where((f) => f.path.endsWith('.dart'));
  
  for (final file in files) {
    String content = file.readAsStringSync();
    if (content.contains('showModalBottomSheet(') && !content.contains('useRootNavigator: true')) {
      content = content.replaceAll(
        'showModalBottomSheet(',
        'showModalBottomSheet(\n      useRootNavigator: true,'
      );
      file.writeAsStringSync(content);
      print('Fixed \${file.path}');
    }
  }
}
